service/src/main/java/dk/dbc/weekresolver/service/WeekResolver.java

```diff
@@ -216,20 +217,24 @@ public WeekResolverResult getWeekCode(LocalDate customDate) throws UnsupportedOp
         LOGGER.debug("======================== BEGIN SHIFTDAY CALCULATION ==================================");
         if (!configuration.getIgnoreClosingDays()) {
 
-            // Step 1: Adjust shiftday, but only if we are not ignoring closing days
-            DayOfWeek shiftDay = configuration.getShiftDay() == null
-                    ? configuration.getShiftDay()
-                    : adjustShiftDay(expectedDate, configuration.getShiftDay(), configuration.getAllowEndOfYear());
+            // Step 1: Adjust shiftday, but only if we are not ignoring closing days and/or shiftday
+            if (configuration.getShiftDay() == null) {
+                LOGGER.debug("No shiftday for this code, date unchanged at {}", expectedDate);
+            } else {
 
-            // Step 2: Is this on or after the shiftday ?
-            if (shiftDay == null || expectedDate.getDayOfWeek().getValue() >= shiftDay.getValue()) {
-                expectedDate = getMonday(expectedDate.plusWeeks(1));
-                LOGGER.debug("Date shifted to monday next week due to shiftday to {}", expectedDate);
+                // Adjust shiftday
+                DayOfWeek shiftDay = adjustShiftDay(expectedDate, configuration.getShiftDay(), configuration.getAllowEndOfYear());
 
-                LOGGER.debug("Checking for closing days");
-                while (!configuration.getIgnoreClosingDays() && (isClosingDay(expectedDate, configuration.getAllowEndOfYear()) || isEasterWeek(expectedDate))) {
-                    expectedDate = expectedDate.plusDays(1);
-                    LOGGER.debug("Date shifted 1 more day due to closing day, easter week or first week of the year to {}", expectedDate);
+                // Step 2: Is this on or after the shiftday ?
+                if (shiftDay == null || expectedDate.getDayOfWeek().getValue() >= shiftDay.getValue()) {
+                    expectedDate = getMonday(expectedDate.plusWeeks(1));
+                    LOGGER.debug("Date shifted to monday next week due to shiftday {} to {}", shiftDay, expectedDate);
```

Comment:
Hmmm : to shiftday {} to {} ?

Reply:
"...Due to shiftday THURSDAY to 2023-05-01..."